### PR TITLE
tests: Export struct fields in the e2e tests

### PR DESCRIPTION
**Pull request by p-offtermatt**
## Description

Closes: Part of #1265 

This PR makes the fields of the relevant structs used in traces exported.
This is necessary preparation for the json format, since JSON marshal/unmarshal has problems with unexported fields, see https://stackoverflow.com/questions/11126793/json-and-dealing-with-unexported-fields.

To review this, please check that the only changes this PR makes are setting the fields of structs and types used in the e2e tests to public. In particular, the fields of all actions need to be exported.
There are also two small refactors that I have marked with comments.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

File: tests/e2e/steps.go [hunk 1]
```diff
@@ -66,15 +66,15 @@ var slashThrottleSteps = concatSteps(
 	stepsStopChain("consu", 2),
 )
 
-var democracySteps = concatSteps(
+var democracyRewardsSteps = concatSteps(
```
Comment (p-offtermatt) [inline tests/e2e/steps.go:69]: Small refactor: The naming was confusing before: democracySteps were the input for democracy-reward, and rewardDenomConsumerSteps were the input for democracy. This makes the mapping of test cases to step names more consistent.

File: tests/e2e/main.go [hunk 2]
```diff
@@ -65,8 +65,8 @@ since causing light client attacks is not implemented.`,
 		},
 		"happy-path":       {testRun: DefaultTestRun(), steps: happyPathSteps, description: "happy path tests"},
 		"changeover":       {testRun: ChangeoverTestRun(), steps: changeoverSteps, description: "changeover tests"},
-		"democracy-reward": {testRun: DemocracyTestRun(true), steps: democracySteps, description: "democracy tests allowing rewards"},
-		"democracy":        {testRun: DemocracyTestRun(false), steps: rewardDenomConsumerSteps, description: "democracy tests"},
+		"democracy-reward": {testRun: DemocracyTestRun(true), steps: democracyRewardsSteps, description: "democracy tests allowing rewards"},
```
Comment (p-offtermatt) [inline tests/e2e/main.go:68]: Small refactor: The step names were mismatched with the test-case names before

File: tests/e2e/actions.go [hunk 3]
```diff
@@ -199,15 +200,16 @@ func (tr TestRun) submitTextProposal(
...
-		`--title`, action.title,
-		`--description`, action.description,
-		`--deposit`, fmt.Sprint(action.deposit)+`stake`,
-		`--from`, `validator`+fmt.Sprint(action.from),
-		`--chain-id`, string(tr.chainConfigs[action.chain].chainId),
-		`--home`, tr.getValidatorHome(action.chain, action.from),
-		`--node`, tr.getValidatorNode(action.chain, action.from),
+		`--title`, action.Title,
+		`--description`, action.Description,
+		`--type`, action.PropType,
```
Comment (insumity) [inline tests/e2e/actions.go:207]: this `--type` seems new, is this expected?

File: tests/e2e/steps_multi_consumer_downtime.go [hunk 4]
```diff
@@ -229,190 +229,190 @@ func stepsMultiConsumerDowntimeFromProvider(consumer1, consumer2 string) []Step
...
-			action: relayPacketsAction{
-				chainA:  chainID("provi"),
-				chainB:  chainID(consumer1),
-				port:    "provider",
-				channel: 0, // consumer1 channel
+			Action: relayPacketsAction{
+				ChainA:  ChainID("provi"),
+				ChainB:  ChainID(consumer1),
+				Port:    "provider",
+				Channel: 0, // consumer1 channel
```
Comment (insumity) [inline tests/e2e/steps_multi_consumer_downtime.go:362]: nit: you made this change before in other files.
s/channel/Channel

File: tests/e2e/steps_multi_consumer_downtime.go [hunk 5]
```diff
@@ -229,190 +229,190 @@ func stepsMultiConsumerDowntimeFromProvider(consumer1, consumer2 string) []Step
...
-			action: relayPacketsAction{
-				chainA:  chainID("provi"),
-				chainB:  chainID(consumer2),
-				port:    "provider",
-				channel: 1, // consumer2 channel
+			Action: relayPacketsAction{
+				ChainA:  ChainID("provi"),
+				ChainB:  ChainID(consumer2),
+				Port:    "provider",
+				Channel: 1, // consumer2 channel
```
Comment (insumity) [inline tests/e2e/steps_multi_consumer_downtime.go:394]: nit: similarly to previous comment

**Review by insumity — APPROVED** [review]
LGTM. Thanks for the "To review this _[...]_ exported." text in your PR.

File: tests/e2e/steps_multi_consumer_downtime.go (hunk 4 above)
Comment (p-offtermatt) [reply tests/e2e/steps_multi_consumer_downtime.go:362]: changed it back to small channel in comments in other places, thanks!

File: tests/e2e/steps_multi_consumer_downtime.go (hunk 5 above)
Comment (p-offtermatt) [reply tests/e2e/steps_multi_consumer_downtime.go:394]: same as previous comment

**Comment by smarshall-spitzbart** [comment]
This PR should be merged directly to main imo, as it will pollute the diff to any future feature branch PR. 

ie. we don't benefit from split PRs if this PR is merged into a feature branch

File: tests/e2e/main.go (hunk 2 above)
Comment (smarshall-spitzbart) [reply tests/e2e/main.go:68]: Nice catch!

**Review by smarshall-spitzbart — APPROVED** [review]
Looks great!

**Pull request merged**